File: Book-server/models/Book.js
const mongoose = require('mongoose');

const ChapterSchema = new mongoose.Schema({
    title: { type: String, required: true },
    order: { type: Number, required: true },
    content: { type: String, required: true },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now },
});

const BookSchema = new mongoose.Schema({
    title: { type: String, required: true },
    author: { type: String, required: true },
    category: { type: String },
    description: { type: String },
    coverUrl: { type: String },
    genre: [String],
    publishedYear: Number,
    pages: Number,
    rating: Number,
    isFavorite: Boolean,
    chapters: [ChapterSchema]
}, { timestamps: true });

module.exports = mongoose.model('Book', BookSchema);

File: app/src/main/java/com/example/book/model/Book.kt
package com.example.book.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Chapter(
    val title: String,
    val content: String,
    val createdAt: String? = null
) : Parcelable

@Parcelize
data class Book(
    val _id: String? = null,
    val title: String,
    val author: String,
    val description: String,
    val genre: List<String> = emptyList(),
    val publishedYear: Int,
    val pages: Int,
    val coverImageUrl: String,
    val isFavorite: Boolean = false,
    val rating: Double = 0.0,
    val chapters: List<Chapter> = emptyList()
) : Parcelable

